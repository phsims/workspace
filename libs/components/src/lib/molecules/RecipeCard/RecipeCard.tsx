import Image from 'next/image';
import DOMPurify from 'isomorphic-dompurify';
import ArrowForwardIosRoundedIcon from '@mui/icons-material/ArrowForwardIosRounded';
import { Card, CardContent, Typography, Box, CardActions } from '@mui/material';
import ButtonLink from '../../atoms/ButtonLink/ButtonLink';

//imort image placeholder

export interface RecipeCardProps {
  id: number;
  image: string;
  title: string;
  summary: string;
}
export function RecipeCard({ image, title, summary, id }: RecipeCardProps) {
  const shortSummery = DOMPurify.sanitize(`${summary.substring(0, 70)}...`);

  return (
    <Card
      sx={{ display: 'flex', flexDirection: 'column', height: '100%' }}
      data-testid={id}
    >
      <Box sx={{ position: 'relative', width: '100%', height: '140px' }}>
        <Image
          placeholder="blur"
          blurDataURL="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mO0t3fxBwACmgETyBRW3gAAAABJRU5ErkJggg=="
          src={image}
          alt={title}
          sizes="400px"
          fill
          style={{
            objectFit: 'cover',
          }}
        />
      </Box>
      <CardContent>
        <Typography gutterBottom variant="h5">
          {title}
        </Typography>
        <Typography
          variant="body2"
          color="text.secondary"
          dangerouslySetInnerHTML={{ __html: shortSummery }}
        />
      </CardContent>

      <CardActions sx={{ mt: 'auto' }}>
        <ButtonLink
          text="Go to recipe"
          link={`recipes/${id}`}
          variant="text"
          size="small"
          endIcon={<ArrowForwardIosRoundedIcon />}
        />
      </CardActions>
    </Card>
  );
}

export default RecipeCard;
