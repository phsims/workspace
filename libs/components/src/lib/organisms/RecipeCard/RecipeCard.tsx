import Image from 'next/image';
import DOMPurify from 'isomorphic-dompurify';
import { Card, CardContent, Typography, Box, CardActions, Button } from '@mui/material';
import ButtonLink from '../../atoms/ButtonLink/ButtonLink';

export interface RecipeCardProps {
  id: number;
  image: string;
  title: string;
  summary: string;
  link:string;
}

export function RecipeCard({ image, title, summary ,link}: RecipeCardProps) {
  const content = DOMPurify.sanitize(`${summary.substring(0, 70)}...`);
  return (
    <Card>
      <Box sx={{ position: 'relative', width: '100%', height: '140px' }}>
        <Image
          src={image}
          alt={title}
          sizes="300px"
          fill
          style={{
            objectFit: 'cover',
          }}
        />
      </Box>
      <CardContent>
        <Typography gutterBottom variant="h5">
          {title}
        </Typography>
        <Typography
          variant="body2"
          color="text.secondary"
          dangerouslySetInnerHTML={{ __html: content }}
        />
      </CardContent>
      <CardActions>
        <ButtonLink text='Get the recipe' link={link} />
      </CardActions>
    </Card>
  );
}

export default RecipeCard;
