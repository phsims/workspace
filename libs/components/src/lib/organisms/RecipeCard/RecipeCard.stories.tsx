import type { Meta, StoryObj } from '@storybook/react';
import { RecipeCard } from './RecipeCard';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { Box } from '@mui/material';

const meta: Meta<typeof RecipeCard> = {
  component: RecipeCard,
  title: '@organisms/RecipeCard',
  decorators: [
    (Story) => (
      <Box sx={{width:345,}}>
        <Story />
      </Box>
    ),
  ],
};
export default meta;
type Story = StoryObj<typeof RecipeCard>;

export const Primary = {
  args: {
    id:222,
    image:'/contemplative-reptile.jpg',
    title:'Lizard',
    summary:'Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging across all continents except Antarctica',
  link:'/link1'
  },
  parameters: {
    nextjs: {
      appDirectory: true,
    },
  },
};

export const Heading: Story = {
  args: {...Primary.args},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Lizard/gi)).toBeInTheDocument();
  },
};
