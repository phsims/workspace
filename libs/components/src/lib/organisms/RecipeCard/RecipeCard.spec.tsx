import { render } from '@testing-library/react';
import { useRouter } from 'next-router-mock';
import mockRouter from 'next-router-mock';

import RecipeCard from './RecipeCard';

const defaultProps = {
  id: 0,
  image: '/test.png',
  title: 'test title',
  summary: 'test summary',
  link: '/test-link',
};

jest.mock('next/navigation', () => require('next-router-mock'));

describe('RecipeCard', () => {
 it('should match snapshot', () => {
    const { baseElement } = render(<RecipeCard {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  })
  it('should render successfully', () => {
    const { baseElement } = render(<RecipeCard {...defaultProps} />);
    expect(baseElement).toBeTruthy();
  });
});
