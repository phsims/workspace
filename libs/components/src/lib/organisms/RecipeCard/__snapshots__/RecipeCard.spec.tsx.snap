// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-1urz026"
      >
        <img
          alt="test title"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="300px"
          src="/_next/image?url=%2Ftest.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ftest.png&w=16&q=75 16w, /_next/image?url=%2Ftest.png&w=32&q=75 32w, /_next/image?url=%2Ftest.png&w=48&q=75 48w, /_next/image?url=%2Ftest.png&w=64&q=75 64w, /_next/image?url=%2Ftest.png&w=96&q=75 96w, /_next/image?url=%2Ftest.png&w=128&q=75 128w, /_next/image?url=%2Ftest.png&w=256&q=75 256w, /_next/image?url=%2Ftest.png&w=384&q=75 384w, /_next/image?url=%2Ftest.png&w=640&q=75 640w, /_next/image?url=%2Ftest.png&w=750&q=75 750w, /_next/image?url=%2Ftest.png&w=828&q=75 828w, /_next/image?url=%2Ftest.png&w=1080&q=75 1080w, /_next/image?url=%2Ftest.png&w=1200&q=75 1200w, /_next/image?url=%2Ftest.png&w=1920&q=75 1920w, /_next/image?url=%2Ftest.png&w=2048&q=75 2048w, /_next/image?url=%2Ftest.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          test title
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          test summary...
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Get the recipe
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
